### PR TITLE
Create geotiff 1.7.1 migration

### LINKED ISSUE #2654: Is a geotiff=1.7.1 migration needed?
### Comment:

I'm looking into dll-hell in the GDAL sphere and I think that geotiff=1.7.1 needs a migration done or its pin updated?  

I see this repo has
```
  geotiff:
    max_pin: x.x.x
...
geotiff:
  - '1.7'
```

I looked at the most recent bot logs and did not find any `geotiff` entries?  Does this need a manual migration or ?

### PR DESCRIPTION
**Pull request by akrherz**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
*  Bumped the build number (if the version is unchanged)
* Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* Ensured the license file is being packaged.

This PR:
1. Updates the pinned version of `geotiff` to use a version without single quotes, maybe this was causing grief before?
2. Creates a manual migration of `geotiff`, since I can not figure out why the bot has not created on.

closes #2654

### CODE REVIEW

**Comment by hmaarrfk** [comment]
This is one recipe that i haven't been rerendering.

**Comment by hmaarrfk** [comment]
Because we've had run exports in the recipe itself, could you also remove the pinnings from this repo. Let them get picked up by the package instead of here.

**Comment by hmaarrfk** [comment]
Because we've had run exports in the recipe itself, could you also remove the pinnings from this repo. Let them get picked up by the package instead of here.

**Comment by akrherz** [comment]
Hi @hmaarrfk, thanks and pardon the ignorance here, but if the version is removed here, will the bot still attempt migrations to repos depending on `geotiff` ?

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -204,8 +204,6 @@ pin_run_as_build:
     max_pin: x
   gdal:
     max_pin: x.x
-  geotiff:
-    max_pin: x.x.x
```
Comment (hmaarrfk) [inline recipe/conda_build_config.yaml:208]: This is what i meant.

**Comment by hmaarrfk** [comment]
It should still attempt it. 

For example, it did suggest we migrate ffmpeg to 5.0 
https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/conda_build_config.yaml#L421
and ffmpeg doesn't have a pinning (we changed it to 4.4 for other reason).

**Comment by akrherz** [comment]
Indeed, sorry, I am fully tracking what you suggested now.

**Comment by hmaarrfk** [comment]
No need to be sorry.

**Comment by isuruf** [comment]
Can you open an issue at https://github.com/lvc/abi-tracker/issues asing about updating libgeotiff report?

**Comment by akrherz** [comment]
> Can you open an issue at https://github.com/lvc/abi-tracker/issues asing about updating libgeotiff report?

Done lvc/abi-tracker#16

**Pull request merged**